More fs2/cats porting

aeons opened the pull request:
Ports the servlet, tomcat and jetty modules to fs2/cats.

The modules compile and the tests pass, but there a few `TODO fs2 port` in there.

Part of #661

aeons commented on jetty/src/test/scala/org/http4s/server/jetty/Issue454.scala:
```diff
@@ -44,6 +44,9 @@ object Issue454 {
+    servletIo    = org.http4s.servlet.NonBlockingServletIo(4096),
+    threadPool   = org.http4s.util.threads.newDefaultFixedThreadPool(
+      4, org.http4s.util.threads.threadFactory(i => s"org.http4s.server.jetty.DefaultExecutor-$i")
+    )
```
I didn't add a `TODO` comment here, since I guess this is fine for testing.

rossabaker replied:
I added a thread pool to `Http4sSpec`'s companion object for these "I just need a thread pool, any old one will do" cases.

rossabaker commented on servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala:
```diff
@@ -83,7 +81,7 @@ class Http4sServlet(service: HttpService,
                             bodyWriter: BodyWriter): Task[Unit] = {
     ctx.addListener(new AsyncTimeoutHandler(request, bodyWriter))
-    val response = Task.fork(serviceFn(request))(threadPool)
+    val response: Task[Response] = Task.start(serviceFn(request))(Strategy.fromExecutor(threadPool)).unsafeRun
```
This `unsafeRun` in the middle looks suspicious.  We have a `Task[Task[Response]]` now, right?  Can we just `.join` or `.flatten` or `.flatMap(identity)` or whatever compiles here?

aeons commented:
Addressed the comments. I tried running the Issue454 thing. It logs an `EofException` but seems to complete normally.

```
16:15:53.579 [qtp97636642-176534] DEBUG org.eclipse.jetty.server.HttpOutput - onError
org.eclipse.jetty.io.EofException: null
	at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:199)
	at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:420)
	at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:313)
	at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:140)
	at org.eclipse.jetty.server.HttpConnection$SendCallback.process(HttpConnection.java:744)
	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:241)
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:224)
	at org.eclipse.jetty.server.HttpConnection.send(HttpConnection.java:518)
	at org.eclipse.jetty.server.HttpChannel.sendResponse(HttpChannel.java:724)
	at org.eclipse.jetty.server.HttpChannel.write(HttpChannel.java:775)
	at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:235)
	at org.eclipse.jetty.server.HttpOutput$AsyncWrite.process(HttpOutput.java:1174)
	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:241)
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:224)
	at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:426)
	at java.io.OutputStream.write(OutputStream.java:75)
	at org.http4s.servlet.NonBlockingServletIo$$anonfun$3.apply(ServletIo.scala:180)
	at org.http4s.servlet.NonBlockingServletIo$$anonfun$3.apply(ServletIo.scala:176)
	at org.http4s.servlet.NonBlockingServletIo$$anonfun$initWriter$2$$anonfun$apply$4$$anonfun$apply$6.apply(ServletIo.scala:234)
	at org.http4s.servlet.NonBlockingServletIo$$anonfun$initWriter$2$$anonfun$apply$4$$anonfun$apply$6.apply(ServletIo.scala:234)
	at fs2.Task$$anonfun$map$1$$anonfun$apply$8$$anonfun$apply$9.apply(Task.scala:47)
...
```

rossabaker commented:
👍 Thanks!
